### ISSUE TITLE
fix: OptimalController premium estimation uses heuristic instead of actuarial pricing

### ISSUE THREAD

**Issue by AlexFiliakov** [AlexFiliakov/Ergodic-Insurance-Limits#1031]
## Problem

In `optimal_control.py`, the `OptimalController._estimate_premium_rate()` method computes insurance premium as a simplified function of the retention level without using any actuarial pricing methodology:

```python
def _estimate_premium_rate(self, retention: float, limit: float) -> float:
    # Heuristic premium estimation
    expected_loss = self._estimate_expected_layer_loss(retention, limit)
    premium = expected_loss * (1 + self.config.expense_ratio + self.config.profit_loading)
    return premium
```

This heuristic:
1. Does not use the `InsurancePricer` or `LayerPricer` classes that exist in the codebase
2. Ignores market cycle effects on pricing
3. Does not account for loss development or IBNR in the premium basis
4. Uses a flat multiplicative loading rather than proper risk-adjusted pricing

### Why This Matters

The HJB optimal control framework determines the **optimal insurance program** by solving the Hamilton-Jacobi-Bellman equation. The premium function `P(retention, limit)` is a critical input — if the premium model is unrealistic, the "optimal" solution will be wrong.

Specifically:
- **Convexity**: Real insurance pricing is convex in coverage (ILFs increase faster than linearly). The heuristic may not capture this, leading to corner solutions.
- **Market cycles**: Premium varies 30-50% across hard/soft markets. Ignoring this means the "optimal" program is only optimal for a single market condition.
- **Layer pricing**: Excess layers price at higher rates per unit of coverage due to increased leverage. A flat loading misses this.
- **Feedback loop**: The whole point of optimal control is to jointly optimize retention/limit decisions with the premium they generate. An inaccurate premium model breaks this feedback.

### Recommended Fix

1. Wire `OptimalController` to use `LayerPricer` for premium estimation
2. Incorporate market cycle state into the premium function (state-dependent pricing)
3. Ensure ILF/LER curves are used for layer-specific pricing
4. Add option to use empirical premium functions from market data

### Affected Files
- `ergodic_insurance/optimal_control.py`
- `ergodic_insurance/insurance_pricing.py`
